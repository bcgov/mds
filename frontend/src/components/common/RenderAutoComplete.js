import React from "react";
import PropTypes from "prop-types";
import { Icon, Input, AutoComplete } from "antd";
import * as Styles from "@/constants/styles";

/**
 * @constant RenderAutoComplete - Ant Design `AutoComplete` component for redux-form.
 *
 */

const propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSelect: PropTypes.func.isRequired,
  data: PropTypes.arrayOf(PropTypes.any).isRequired,
  placeholder: PropTypes.string,
  iconColor: PropTypes.string,
};

const defaultProps = {
  placeholder: "",
  iconColor: Styles.COLOR.violet,
};

const RenderAutoComplete = (props) => (
  <AutoComplete
    defaultActiveFirstOption={false}
    notFoundContent="Not Found"
    allowClear
    dropdownMatchSelectWidth
    backfill
    style={{ width: "100%" }}
    dataSource={props.data}
    placeholder={props.placeholder}
    filterOption={(input, option) =>
      option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
    }
    onSelect={props.handleSelect}
    onChange={props.handleChange}
  >
    <Input
      autoComplete="off"
      id="search"
      suffix={<Icon type="search" className="icon-sm" style={{ color: props.iconColor }} />}
    />
  </AutoComplete>
);

RenderAutoComplete.propTypes = propTypes;
RenderAutoComplete.defaultProps = defaultProps;

export default RenderAutoComplete;
