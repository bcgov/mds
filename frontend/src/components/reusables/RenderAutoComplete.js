import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { Icon, Input, AutoComplete } from 'antd';

const propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSelect: PropTypes.func.isRequired,
  data: PropTypes.array.isRequired,
  placeholder: PropTypes.string
};

/**
 * Ant Design `AutoComplete` component for redux-form.
 * 
 */
const RenderAutoComplete = ({
  data,
  placeholder,
  handleChange,
  handleSelect
}) => (
    <AutoComplete
      defaultActiveFirstOption={false}
      notFoundContent={'Not Found'}
      allowClear
      dropdownMatchSelectWidth={true}
      size="large"
      backfill={true}
      style={{ width: '100%' }}
      dataSource={data}
      placeholder={placeholder}
      optionLabelProp="value"
      filterOption={(input, option) => option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0}
      onSelect={handleSelect}
      onChange={handleChange}
    >
      <Input suffix={<Icon type="search" style={{ color: '#537C52', fontSize: 20 }} />} />
    </AutoComplete>
  );

RenderAutoComplete.propTypes = propTypes;

export default RenderAutoComplete;
