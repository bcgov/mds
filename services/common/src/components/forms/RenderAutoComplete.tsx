import React from "react";
import PropTypes from "prop-types";
import { Select, Form } from "antd";

/**
 * @constant RenderAutoComplete - Ant Design `AutoComplete` component for redux-form.
 *
 */

const propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSelect: PropTypes.func.isRequired,
  data: PropTypes.arrayOf(PropTypes.any).isRequired,
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  meta: PropTypes.objectOf(PropTypes.any),
  input: PropTypes.objectOf(PropTypes.any),
  selected: PropTypes.objectOf(PropTypes.any),
};

const defaultProps = {
  placeholder: "",
  disabled: false,
  meta: {},
  input: null,
  selected: undefined,
};

const RenderAutoComplete = (props) => {
  return (
    <Form.Item
      label={JSON.stringify(props.selected)}
      validateStatus={
        props.meta.touched ? (props.meta.error && "error") || (props.meta.warning && "warning") : ""
      }
      help={
        props.meta.touched &&
        ((props.meta.error && <span>{props.meta.error}</span>) ||
          (props.meta.warning && <span>{props.meta.warning}</span>))
      }
    >
      <Select
        showSearch
        virtual={false}
        defaultActiveFirstOption={false}
        notFoundContent="Not Found"
        allowClear
        dropdownMatchSelectWidth
        defaultValue={props.input ? props.input.value : undefined}
        value={props.input ? props.input.value : undefined}
        style={{ width: "100%" }}
        options={props.data}
        placeholder={props.placeholder}
        filterOption={(input, option) =>
          option.label
            .toString()
            .toLowerCase()
            .indexOf(input.toLowerCase()) >= 0
        }
        disabled={props.disabled}
        onChange={props.input ? props.input.onChange : undefined}
        onSelect={props.handleSelect}
        onSearch={(event) => {
          props.handleChange(event);
          if (props.input) {
            props.input.onChange(event);
          }
        }}
      />
    </Form.Item>
  );
};

RenderAutoComplete.propTypes = propTypes;
RenderAutoComplete.defaultProps = defaultProps;

export default RenderAutoComplete;
