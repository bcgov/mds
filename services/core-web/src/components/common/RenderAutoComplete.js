import React from "react";
import PropTypes from "prop-types";
import { Form, AutoComplete } from "antd";
import * as Styles from "@/constants/styles";

/**
 * @constant RenderAutoComplete - Ant Design `AutoComplete` component for redux-form.
 *
 */

const propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSelect: PropTypes.func.isRequired,
  data: PropTypes.arrayOf(PropTypes.any).isRequired,
  id: PropTypes.string,
  placeholder: PropTypes.string,
  defaultValue: PropTypes.string,
  label: PropTypes.string,
  iconColor: PropTypes.string,
  disabled: PropTypes.bool,
  meta: PropTypes.objectOf(PropTypes.any),
  input: PropTypes.objectOf(PropTypes.any),
};

const defaultProps = {
  id: "search",
  placeholder: "",
  defaultValue: "",
  label: "",
  iconColor: Styles.COLOR.violet,
  disabled: false,
  meta: {},
  input: null,
};

const RenderAutoComplete = (props) => {
  return (
    <Form.Item
      label={props.label}
      validateStatus={
        props.meta.touched ? (props.meta.error && "error") || (props.meta.warning && "warning") : ""
      }
      help={
        props.meta.touched &&
        ((props.meta.error && <span>{props.meta.error}</span>) ||
          (props.meta.warning && <span>{props.meta.warning}</span>))
      }
    >
      <AutoComplete
        defaultActiveFirstOption={false}
        notFoundContent="Not Found"
        allowClear
        dropdownMatchSelectWidth
        backfill
        defaultValue={props.defaultValue}
        style={{ width: "100%" }}
        dataSource={props.data}
        placeholder={props.placeholder}
        filterOption={(input, option) =>
          option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
        }
        disabled={props.disabled}
        {...props.input}
        onSelect={props.handleSelect}
        onChange={(event) => {
          props.handleChange(event);
          if (props.input) {
            props.input.onChange(event);
          }
        }}
      />
    </Form.Item>
  );
};

RenderAutoComplete.propTypes = propTypes;
RenderAutoComplete.defaultProps = defaultProps;

export default RenderAutoComplete;
