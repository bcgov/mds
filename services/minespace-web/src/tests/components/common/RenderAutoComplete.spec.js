import React from "react";
import { shallow } from "enzyme";
import RenderAutoComplete from "@/components/common/RenderAutoComplete";
import { PARTY } from "@/tests/mocks/dataMocks";

let props = {};
let dispatchProps = {};

const setupProps = () => {
  props = {
    id: "1",
    input: {},
    label: "",
    type: "",
    meta: {
      touched: false,
      error: false,
      warning: false,
    },
    data: [],
    option: {},
  };
};

const setupDispatchProps = () => {
  dispatchProps = {
    handleChange: jest.fn(),
    handleSelect: jest.fn(),
  };
};

beforeEach(() => {
  setupProps();
  setupDispatchProps();
});

describe("RenderAutoComplete", () => {
  it("renders properly", () => {
    props.data = PARTY.partyIds;
    props.option = PARTY.parties;
    const wrapper = shallow(<RenderAutoComplete {...props} {...dispatchProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
