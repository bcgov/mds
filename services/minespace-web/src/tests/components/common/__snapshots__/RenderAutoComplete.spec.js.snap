// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderAutoComplete renders properly 1`] = `
<FormItem
  hasFeedback={false}
  help={false}
  validateStatus=""
>
  <ForwardRef(InternalSelect)
    allowClear={true}
    defaultActiveFirstOption={false}
    disabled={false}
    dropdownMatchSelectWidth={true}
    filterOption={[Function]}
    notFoundContent="Not Found"
    onSearch={[Function]}
    onSelect={[MockFunction]}
    options={
      Array [
        "18133c75-49ad-4101-85f3-a43e35ae989a",
        "18145c75-49ad-0101-85f3-a43e45ae989a",
      ]
    }
    placeholder=""
    showSearch={true}
    style={
      Object {
        "width": "100%",
      }
    }
    virtual={false}
  />
</FormItem>
`;
